kafka_utils/kafka_cluster_manager/cmds/command.py

```diff
@@ -248,19 +257,22 @@ def _extract_actions_unique_topics(self, movement_counts, max_movements):
         # Create reduced assignment minimizing duplication of topics
         extracted_actions = []
         curr_movements = 0
+        curr_size = 0
         action_available = True
-        while curr_movements < max_movements and action_available:
+        while curr_movements < max_movements and curr_size < max_movement_size and action_available:
```

Comment:
To be on safer side, should we at least allow 1 partition for movement if all the partitions exceed the given size.
Another alternate is to through a warning/error log if such case arises.

Reply:
Yeah, this is a little tricky since we want to have a good value. The issue arises when clusters have high variance in partition size. In these situations, if we go by the largest partition in the cluster, we won't be able to properly utilize the `max_movement_size` well. But if we don't we'll get to a state where we cannot make progress.

I think adding an `auto-max-movement-size` like in rebalance and setting that to the size of the largest partition, and then printing a warning if the user specified a `max-movement-size` that is less than the size of the largest partition would be enough.

Reply:
Rebalance exits 1 when the max movement size is lower than the biggest partition in the cluster. I think we should do the same in decommission. For decommission failing fast when the process cannot be completed is even more important than rebalance. Perhaps as an optimization we can fail if the max-movement-size is smaller than the largest partition to decommssion.

Reply:
++ That makes a lot of sense. Having the optimization seems doable as well, as we can just iterate through the partitions of the brokers that are to be decommissioned.